tilespec.ts

```diff
@@ -1,54 +1,213 @@
-const MAX_TUPLE_MEMBER_VALUE = Number.MAX_SAFE_INTEGER;
-
 /**
  * TileSpec represents a rectangular area on display by means of specifying a
  * number of evenly spaced tiles and two corners.
  */
 export class TileSpec {
     readonly gridWidth: number;
     readonly gridHeight: number;
-    readonly luc: XY;
-    readonly rdc: XY;
+    readonly luc: TupleHolder;
```

Comment:
We should probably rename these variables to leftUpper and rightLower or something.